src/db/models/endpoints.js

```diff
@@ -99,14 +173,16 @@ endpointSchema.pre('save', async function (next) {
 
   await EndpointModel.find({
     'endpoint.pattern': {$regex: endpointMatchingRegex}
-  }).then(result => {
+  }).then(async result => {
     if (result.length > 0) {
       const error = new Error(
         `Duplicate error: regex created from endpoint pattern ${endpoint.endpoint.pattern} for matching requests already exists`
       )
       return next(error)
     }
-
+    if (!this.validateSync()) {
+      await kafka.subscribeTopicToConsumer(endpoint)
+    }
```

Comment:
I think you need to wrap this in a try catch block

Reply:
Function is used in a function that has the try/catch logic